### PR TITLE
Change the default 'make server_run' command to always rebuild on swagger change

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

Previously in https://github.com/transcom/mymove/pull/866 we added `make server_generate_and_run`.  This PR makes it the default command for running the server.  The hope is to reduce the number of errors devs run across when switching branches or while rapidly developing on the API.

This change modifies the makefile to do this:

- `make server_generate_and_run` -> `make server_run`
- `make server_run` -> `make server_run_default`

I also add `entr` as a project dependency. 

## Reviewer Notes

Could this possibly cause issues in other parts of our infrastructure or CI/CD pipeline?

## Setup

Try:

```sh
./bin/prereqs
```

and

```sh
make server_run
```

## Code Review Verification Steps

* [x] Request review from a member of a different team.

### CODE REVIEW

File: bin/prereqs [hunk 1]
```diff
@@ -22,6 +22,7 @@ has pre-commit "brew install pre-commit"
 has shellcheck "brew install shellcheck"
 has docker "Get Docker CE for Mac from https://download.docker.com/mac/stable/Docker.dmg"
 has psql "brew install postgresql"
+has entr "brew install entr"
```
Comment (chrisgilmerproj) [inline bin/prereqs]: I'm not sure I actually like this change.  `entr` itself isn't a prereq for running our code in production, its only required for running in development.  Would it be possible to solve this by having two files:

- `bin/prereqs-prod` (used by docker files in CI/CD pipeline)
- `bin/prereqs` (calls `bin/prereqs-prod` and includes dev-only requirements)

What do we think?

Comment (jim) [reply bin/prereqs]: I think separating dev-only and prod dependencies in this way is a good idea.

Comment (stangah) [reply bin/prereqs]: Down on line ~32 is a pseudo-dev-only prereq block

Comment (chrisgilmerproj) [reply bin/prereqs]: Oh, that's exactly what I wanted.  Thank you, I'll move it down to that area.

**Pull request merged**